mosquitto: udpate to 1.4.8

mufeng opened the pull request:
Due to 1.4 version has some memory leak issue, update to 1.4.8 in hope
of solving them.

tripzero commented on recipes-devtools/mraa/mraa.inc:
```diff
@@ -14,12 +14,36 @@ FILES_${PN}-doc += "${datadir}/mraa/examples/"
+FILES_${PN}_append = "${@' ${libdir}/../lib/*' if '${MACHINE}' == 'intel-baytrail-64' else ''}"
+
+# include .so symlinks in main package
+FILES_${PN}_append = "${@' ${libdir}/../lib64/*.so' if '${MACHINE}' == 'intel-baytrail-64' else ' ${libdir}/../lib/*.so'}"
```
should this be checking for ${TARGET_ARCH} instead of ${MACHINE}?

tripzero commented on recipes-devtools/mraa/mraa.inc:
```diff
@@ -14,12 +14,36 @@ FILES_${PN}-doc += "${datadir}/mraa/examples/"
+set (JAVA_JVM_LIBRARY ${JAVA_HOME}/jre/lib/amd64/libjvm.so CACHE FILEPATH \"path to JVM\" FORCE)
+" >> ${WORKDIR}/toolchain.cmake
+}
+
```
Should this go in upstream mraa's CMake configuration?

andreivasiliu2211 commented:
This recipe uses cmake.bbclass for cmake based projects. It relies on the existance of a file, toolchain.cmake, which will be used by cmake to set the options for cross-compilation. This file is created in cmake.bbclass and I append to it the options for Java. It cannot be done in mraa's Cmake configuration, because it must be done inside this file, that is generated by Yocto.

tripzero commented:
@andreivasiliu2211 , okay, that makes sense.  Should it be done in upstream poky then?  The java variables are missing from cmake.bbclass.

tripzero commented:
@andreivasiliu2211 do you think it's possible to update this to use mraa 0.10?

andreivasiliu2211 commented:
Yes. I will update it tomorrow.

andreivasiliu2211 commented:
Updated with TARGET_ARCH instead of MACHINE.

tripzero commented on recipes-devtools/mraa/mraa_0.9.6.bb:
```diff
@@ -1,3 +1,3 @@
 require mraa.inc
 
-SRC_URI = "git://github.com/intel-iot-devkit/mraa.git;protocol=git;rev=d430b5c2b5b282e40bc705a28158b3d682d722f5"
+SRC_URI = "git://github.com/intel-iot-devkit/mraa.git;protocol=git;rev=d8024bb516bf2d895eaff6f92115835d7ef704a2"
```
is this still 0.9.6?  Should we rename file to mraa_0.10.0.bb if this rev is 0.10?

mircea replied:
MRAA 0.10.1 does not build in Yocto with Node & Python bindings enabled. This is the latest version that has all bindings compile in Yocto. We will push an updated recipe once a fix is out for MRAA.

mufeng committed:
mosquitto: udpate to 1.4.8

Due to 1.4 version has some memory leak issue, update to 1.4.8 in hope
of solving them.

tripzero commented:
@mufeng does the new version need c-ares?  Or can it be built without it (optional component)?

tripzero commented:
Looks like it doesn't want to build without c-ares, which in Ostro is currently not whitelisted.  I need to check to see what the best way of making that whitelisted is before merging to the ostroproject repo.

tripzero commented:
It may be better to submit PR directly to the ostroproject repo: https://github.com/ostroproject/meta-intel-iot-middleware
